src/main/java/com/example/petinnoflightandairlines/service/AirportServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package com.example.petinnoflightandairlines.service;
+
+import com.example.petinnoflightandairlines.dto.AirportDTO;
+import com.example.petinnoflightandairlines.dto.AirportSearchCriteria;
+import com.example.petinnoflightandairlines.mapper.AirportMapper;
+import com.example.petinnoflightandairlines.model.Airport;
+import com.example.petinnoflightandairlines.repository.AirportRepository;
+import com.example.petinnoflightandairlines.specifications.AirportSpecification;
+import io.micrometer.common.util.StringUtils;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class AirportServiceImpl implements AirportService {
+
+    private final AirportRepository airportRepository;
+
+    private final AirportMapper airportMapper;
+
+    @Override
+    public AirportDTO addAirport(@Valid AirportDTO airportDTO) {
+
+        checkAirportCodes(airportDTO);
+        Airport airport = airportMapper.convertAirportDTOToAirport(airportDTO);
+
+        Airport savedAirport = saveAirport(airport);
+        return airportMapper.convertAirportToAirportDTO(savedAirport);
+    }
+
+    @Override
+    public List<AirportDTO> getAirports(AirportSearchCriteria criteria) {
+
+        List<Airport> airports;
+
+        if (criteria.isEmpty()) {
+            airports = airportRepository.findAll();
+        } else {
+            airports = getAirportsBy(criteria);
+        }
+
+        if (airports.isEmpty()) {
+            throw new RuntimeException("There are not airports with such fields");
```

Comment:
1) I think in this case we just need to send empty list
2) let's use Pageable result for this method